lib/backend/k8s/k8s.go

```diff
@@ -655,3 +659,38 @@ func (c *KubeClient) getHostConfig(k model.HostConfigKey) (*model.KVPair, error)
 func (c *KubeClient) listHostConfig(l model.HostConfigListOptions) ([]*model.KVPair, error) {
 	return []*model.KVPair{}, nil
 }
+
+func (c *KubeClient) listNodes(k model.NodeListOptions) ([]*model.KVPair, error) {
+	nodes, err := c.clientSet.Nodes().List(kapiv1.ListOptions{})
+	if err != nil {
+		k8sErrorToCalico(err, k)
+	}
+
+	kvp := []*model.KVPair{}
+	for _, node := range nodes.Items {
+		n, err := c.getNode(model.NodeKey{Hostname: node.Name})
```

Comment:
This is going to trigger a whole bunch of API queries to get each node (potentially thousands).

The Nodes().List() call will already return all the nodes, so we should just use those :)